src/index/index.h

```diff
@@ -0,0 +1,50 @@
+#ifndef __INDEX_H__
+#define __INDEX_H__
+
+#include <assert.h>
+#include "../redismodule.h"
+#include "../graph/graph_entity.h"
+#include "../stores/store.h"
+#include "../util/skiplist.h"
+#include "../parser/ast.h"
+#include "../graph/node.h"
+#include "../filter_tree/filter_tree.h"
+#include "../parser/grammar.h" // required for the definition of filter operations (LT, GT, etc)
+#include "../arithmetic/tuples_iter.h"
+
+#define INDEX_PREFIX "redis_graph_INDEX"
+
+#define Index_DEPLETED (GrB_Index)SKIPLIST_DEPLETED
+
+typedef TuplesIter IndexIterator; // TODO unused
+typedef skiplistIterator IndexCreateIter;
```

Comment:
An Index is now stored as a skiplist, but `NewIndexScan` converts it into a matrix, and the scan operation uses a `TuplesIter`.

I was thinking that this duality might be confusing, and that as such we could have Create and Scan iters, but I did not implement that completely (and now that I think about it, I don't think it would actually be less confusing).